source/rust_verify/src/rust_to_vir_expr.rs

```diff
@@ -1288,6 +1288,64 @@ pub(crate) fn expr_to_vir_with_adjustments<'tcx>(
     }
 }
 
+pub(crate) fn expr_cast_enum_int_to_vir<'tcx>(
+    bctx: &BodyCtxt<'tcx>,
+    expr: &'tcx Expr<'tcx>,
+    mk_expr: impl Fn(ExprX) -> Result<vir::ast::Expr, vir::messages::Message>,
+    current_modifier: ExprModifier,
+) -> Result<vir::ast::Expr, VirErr> {
+    let tcx = bctx.ctxt.tcx;
+    let ty = bctx.types.node_type(expr.hir_id);
+    let expr_vir = expr_to_vir(bctx, expr, current_modifier)?;
```

Comment:
Since the callers of `expr_cast_enum_int_to_vir` already have `source_vir`, can we just pass that in instead of running `expr_to_vir` again here?